diff --git a/src/index.d.ts b/src/index.d.ts
index 03c6b661930b89457790b49e23b0edd34d8d368b..cad47657dce4e77689d3e70795af513df622580d 100644
--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -78,7 +78,10 @@ declare namespace React {
 	 */
 	type JSXElementConstructor<P> =
 		| ((props: P, context?: any) => ReactNode)
-		| (new (props: P, context?: any) => Component<any, any>);
+		| (new (
+				props: P,
+				context?: any,
+		  ) => Component<any, any>);

 	/**
 	 * A readonly ref container where {@link current} cannot be mutated.
@@ -274,7 +277,10 @@ declare namespace React {
 	 * const element: ReactElement = <frame />;
 	 * ```
 	 */
-	interface ReactElement<P = any, T extends string | JSXElementConstructor<any> = string | JSXElementConstructor<any>> {
+	interface ReactElement<
+		P = any,
+		T extends string | JSXElementConstructor<any> = string | JSXElementConstructor<any>,
+	> {
 		type: T;
 		props: P;
 		key: string | undefined;
@@ -286,11 +292,14 @@ declare namespace React {
 	> extends ReactElement<P, Exclude<T, number>> {}

 	interface FunctionComponentElement<P> extends ReactElement<P, FunctionComponent<P>> {
-		ref?: ("ref" extends keyof P ? (P extends { ref?: infer R | undefined } ? R : never) : never) | undefined;
+		ref?:
+			| ("ref" extends keyof P ? (P extends { ref?: infer R | undefined } ? R : never) : never)
+			| undefined;
 	}

 	type CElement<P, T extends Component<P, ComponentState>> = ComponentElement<P, T>;
-	interface ComponentElement<P, T extends Component<P, ComponentState>> extends ReactElement<P, ComponentClass<P>> {
+	interface ComponentElement<P, T extends Component<P, ComponentState>>
+		extends ReactElement<P, ComponentClass<P>> {
 		ref?: LegacyRef<T> | undefined;
 	}

@@ -329,7 +338,11 @@ declare namespace React {
 		props?: (ClassAttributes<ClassicComponent<P, ComponentState>> & P) | undefined,
 		...children: ReactNode[]
 	): CElement<P, ClassicComponent<P, ComponentState>>;
-	function createElement<P extends {}, T extends Component<P, ComponentState>, C extends ComponentClass<P>>(
+	function createElement<
+		P extends {},
+		T extends Component<P, ComponentState>,
+		C extends ComponentClass<P>,
+	>(
 		type: ClassType<P, T, C>,
 		props?: (ClassAttributes<T> & P) | undefined,
 		...children: ReactNode[]
@@ -755,7 +768,10 @@ declare namespace React {
 		// Also, the ` | S` allows intellisense to not be dumbisense
 		setState<K extends keyof S>(
 			state:
-				| ((prevState: Readonly<S>, props: Readonly<P>) => Pick<MapToNone<S>, K> | MapToNone<S> | undefined)
+				| ((
+						prevState: Readonly<S>,
+						props: Readonly<P>,
+				  ) => Pick<MapToNone<S>, K> | MapToNone<S> | undefined)
 				| (Pick<MapToNone<S>, K> | MapToNone<S> | undefined),
 			callback?: () => void,
 		): void;
@@ -934,7 +950,10 @@ declare namespace React {
 	 *
 	 * @see {@link ForwardRefRenderFunction}
 	 */
-	type ForwardedRef<T> = ((instance: T | undefined) => void) | MutableRefObject<T | undefined> | undefined;
+	type ForwardedRef<T> =
+		| ((instance: T | undefined) => void)
+		| MutableRefObject<T | undefined>
+		| undefined;

 	/**
 	 * The type of the function passed to {@link forwardRef}. This is considered different
@@ -1038,7 +1057,10 @@ declare namespace React {
 	 * for more info.
 	 */
 	type ClassType<P, T extends Component<P, ComponentState>, C extends ComponentClass<P>> = C &
-		(new (props: P, context?: any) => T);
+		(new (
+			props: P,
+			context?: any,
+		) => T);

 	//
 	// Component Specs and Lifecycle
@@ -1047,7 +1069,9 @@ declare namespace React {
 	// This should actually be something like `Lifecycle<P, S> | DeprecatedLifecycle<P, S>`,
 	// as React will _not_ call the deprecated lifecycle methods if any of the new lifecycle
 	// methods are present.
-	interface ComponentLifecycle<P, S, SS = any> extends NewLifecycle<P, S, SS>, DeprecatedLifecycle<P, S> {
+	interface ComponentLifecycle<P, S, SS = any>
+		extends NewLifecycle<P, S, SS>,
+			DeprecatedLifecycle<P, S> {
 		/**
 		 * Called immediately after a component is mounted. Setting state here will trigger re-rendering.
 		 */
@@ -1062,7 +1086,11 @@ declare namespace React {
 		 * If false is returned, {@link Component.render}, `componentWillUpdate`
 		 * and `componentDidUpdate` will not be called.
 		 */
-		shouldComponentUpdate?(nextProps: Readonly<P>, nextState: Readonly<S>, nextContext: any): boolean;
+		shouldComponentUpdate?(
+			nextProps: Readonly<P>,
+			nextState: Readonly<S>,
+			nextContext: any,
+		): boolean;
 		/**
 		 * Called immediately before a component is destroyed. Perform any necessary cleanup in this method, such as
 		 * cancelled network requests, or cleaning up any DOM elements created in `componentDidMount`.
@@ -1209,7 +1237,11 @@ declare namespace React {
 		 * @see {@link https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#reading-dom-properties-before-an-update}
 		 * @see {@link https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#gradual-migration-path}
 		 */
-		UNSAFE_componentWillUpdate?(nextProps: Readonly<P>, nextState: Readonly<S>, nextContext: any): void;
+		UNSAFE_componentWillUpdate?(
+			nextProps: Readonly<P>,
+			nextState: Readonly<S>,
+			nextContext: any,
+		): void;
 	}

 	interface Mixin<P, S> extends ComponentLifecycle<P, S> {
@@ -1357,7 +1389,9 @@ declare namespace React {
 	 * type MyComponentPropsWithRef = React.ComponentPropsWithRef<typeof MyComponent>;
 	 * ```
 	 */
-	type ComponentPropsWithRef<T extends ElementType> = T extends new (props: infer P) => Component<any, any>
+	type ComponentPropsWithRef<T extends ElementType> = T extends new (
+		props: infer P,
+	) => Component<any, any>
 		? PropsWithoutRef<P> & RefAttributes<InstanceType<T>>
 		: PropsWithRef<ComponentProps<T>>;
 	/**
@@ -1376,7 +1410,9 @@ declare namespace React {
 	 * type MyComponentPropsWithRef = React.CustomComponentPropsWithRef<typeof MyComponent>;
 	 * ```
 	 */
-	type CustomComponentPropsWithRef<T extends ComponentType> = T extends new (props: infer P) => Component<any, any>
+	type CustomComponentPropsWithRef<T extends ComponentType> = T extends new (
+		props: infer P,
+	) => Component<any, any>
 		? PropsWithoutRef<P> & RefAttributes<InstanceType<T>>
 		: T extends (props: infer P, context?: any) => ReactNode
 			? PropsWithRef<P>
@@ -1407,16 +1443,19 @@ declare namespace React {
 	 */
 	type ComponentPropsWithoutRef<T extends ElementType> = PropsWithoutRef<ComponentProps<T>>;

-	type ComponentRef<T extends ElementType> =
-		T extends NamedExoticComponent<ComponentPropsWithoutRef<T> & RefAttributes<infer Method>>
+	type ComponentRef<T extends ElementType> = T extends NamedExoticComponent<
+		ComponentPropsWithoutRef<T> & RefAttributes<infer Method>
+	>
+		? Method
+		: ComponentPropsWithRef<T> extends RefAttributes<infer Method>
 			? Method
-			: ComponentPropsWithRef<T> extends RefAttributes<infer Method>
-				? Method
-				: never;
+			: never;

 	// will show `Memo(${Component.displayName || Component.name})` in devtools by default,
 	// but can be given its own specific name
-	type MemoExoticComponent<T extends ComponentType<any>> = NamedExoticComponent<CustomComponentPropsWithRef<T>> & {
+	type MemoExoticComponent<T extends ComponentType<any>> = NamedExoticComponent<
+		CustomComponentPropsWithRef<T>
+	> & {
 		readonly type: T;
 	};

@@ -1440,14 +1479,21 @@ declare namespace React {
 	 */
 	function memo<P extends object>(
 		Component: FunctionComponent<P>,
-		propsAreEqual?: (prevProps: Readonly<PropsWithChildren<P>>, nextProps: Readonly<PropsWithChildren<P>>) => boolean,
+		propsAreEqual?: (
+			prevProps: Readonly<PropsWithChildren<P>>,
+			nextProps: Readonly<PropsWithChildren<P>>,
+		) => boolean,
 	): NamedExoticComponent<P>;
 	function memo<T extends ComponentType<any>>(
 		Component: T,
-		propsAreEqual?: (prevProps: Readonly<ComponentProps<T>>, nextProps: Readonly<ComponentProps<T>>) => boolean,
+		propsAreEqual?: (
+			prevProps: Readonly<ComponentProps<T>>,
+			nextProps: Readonly<ComponentProps<T>>,
+		) => boolean,
 	): MemoExoticComponent<T>;

-	interface LazyExoticComponent<T extends ComponentType<any>> extends ExoticComponent<CustomComponentPropsWithRef<T>> {
+	interface LazyExoticComponent<T extends ComponentType<any>>
+		extends ExoticComponent<CustomComponentPropsWithRef<T>> {
 		readonly _result: T;
 	}

@@ -1471,7 +1517,9 @@ declare namespace React {
 	 * const MyComponent = lazy(() => import("./MyComponent"));
 	 * ```
 	 */
-	function lazy<T extends ComponentType<any>>(load: () => Promise<{ default: T }>): LazyExoticComponent<T>;
+	function lazy<T extends ComponentType<any>>(
+		load: () => Promise<{ default: T }>,
+	): LazyExoticComponent<T>;

 	//
 	// React Hooks
@@ -1537,7 +1585,9 @@ declare namespace React {
 	 * @version 16.8.0
 	 * @see {@link https://react.dev/reference/react/useContext}
 	 */
-	function useContext<T>(context: Context<T> /*, (not public API) observedBits?: number|boolean */): T;
+	function useContext<T>(
+		context: Context<T> /*, (not public API) observedBits?: number|boolean */,
+	): T;
 	/**
 	 * Returns a stateful value, and a function to update it.
 	 *
@@ -1552,7 +1602,9 @@ declare namespace React {
 	 * @version 16.8.0
 	 * @see {@link https://react.dev/reference/react/useState}
 	 */
-	function useState<S = undefined>(): LuaTuple<[S | undefined, Dispatch<SetStateAction<S | undefined>>]>;
+	function useState<S = undefined>(): LuaTuple<
+		[S | undefined, Dispatch<SetStateAction<S | undefined>>]
+	>;
 	/**
 	 * An alternative to `useState`.
 	 *
@@ -1717,7 +1769,11 @@ declare namespace React {
 	 * @version 16.8.0
 	 * @see {@link https://react.dev/reference/react/useImperativeHandle}
 	 */
-	function useImperativeHandle<T, R extends T>(ref: Ref<T> | undefined, init: () => R, deps?: DependencyList): void;
+	function useImperativeHandle<T, R extends T>(
+		ref: Ref<T> | undefined,
+		init: () => R,
+		deps?: DependencyList,
+	): void;
 	// I made 'inputs' required here and in useMemo as there's no point to memoizing without the memoization key
 	// useCallback(X) is identical to just using X, useMemo(() => Y) is identical to just using Y.
 	/**
@@ -1922,7 +1978,9 @@ declare namespace React {
 // so boolean is only resolved for T = any
 type IsExactlyAny<T> = boolean extends (T extends never ? true : false) ? true : false;

-type ExactlyAnyPropertyKeys<T> = { [K in keyof T]: IsExactlyAny<T[K]> extends true ? K : never }[keyof T];
+type ExactlyAnyPropertyKeys<T> = {
+	[K in keyof T]: IsExactlyAny<T[K]> extends true ? K : never;
+}[keyof T];
 type NotExactlyAnyPropertyKeys<T> = Exclude<keyof T, ExactlyAnyPropertyKeys<T>>;

 // Try to resolve ill-defined props like for JS users: props can be any, or sometimes objects with properties of type any
@@ -1964,242 +2022,3 @@ type ReactManagedAttributes<C, P> = C extends { propTypes: infer T; defaultProps
 		: C extends { defaultProps: infer D }
 			? Defaultize<P, D>
 			: P;
-
-declare global {
-	namespace JSX {
-		type ElementType = string | React.JSXElementConstructor<any>;
-		interface Element extends React.ReactElement<any, any> {}
-		interface ElementClass extends React.Component<any> {
-			render(): React.ReactNode;
-		}
-		interface ElementAttributesProperty {
-			props: {};
-		}
-		interface ElementChildrenAttribute {
-			children: {};
-		}
-
-		// We can't recurse forever because `type` can't be self-referential;
-		// let's assume it's reasonable to do a single React.lazy() around a single React.memo() / vice-versa
-		type LibraryManagedAttributes<C, P> = C extends
-			| React.MemoExoticComponent<infer T>
-			| React.LazyExoticComponent<infer T>
-			? T extends React.MemoExoticComponent<infer U> | React.LazyExoticComponent<infer U>
-				? ReactManagedAttributes<U, P>
-				: ReactManagedAttributes<T, P>
-			: ReactManagedAttributes<C, P>;
-
-		interface IntrinsicAttributes extends React.Attributes {}
-		interface IntrinsicClassAttributes<T> extends React.ClassAttributes<T> {}
-
-		interface IntrinsicElements {
-			accessory: React.InstanceProps<Accessory>;
-			accessorydescription: React.InstanceProps<AccessoryDescription>;
-			accoutrement: React.InstanceProps<Accoutrement>;
-			actor: React.InstanceProps<Actor>;
-			adgui: React.InstanceProps<AdGui>;
-			adportal: React.InstanceProps<AdPortal>;
-			aircontroller: React.InstanceProps<AirController>;
-			alignorientation: React.InstanceProps<AlignOrientation>;
-			alignposition: React.InstanceProps<AlignPosition>;
-			angularvelocity: React.InstanceProps<AngularVelocity>;
-			animation: React.InstanceProps<Animation>;
-			animationconstraint: React.InstanceProps<AnimationConstraint>;
-			animationcontroller: React.InstanceProps<AnimationController>;
-			animationrigdata: React.InstanceProps<AnimationRigData>;
-			animator: React.InstanceProps<Animator>;
-			archandles: React.InstanceProps<ArcHandles>;
-			atmosphere: React.InstanceProps<Atmosphere>;
-			attachment: React.InstanceProps<Attachment>;
-			audioanalyzer: React.InstanceProps<AudioAnalyzer>;
-			audiochorus: React.InstanceProps<AudioChorus>;
-			audiocompressor: React.InstanceProps<AudioCompressor>;
-			audiodeviceinput: React.InstanceProps<AudioDeviceInput>;
-			audiodeviceoutput: React.InstanceProps<AudioDeviceOutput>;
-			audiodistortion: React.InstanceProps<AudioDistortion>;
-			audioecho: React.InstanceProps<AudioEcho>;
-			audioemitter: React.InstanceProps<AudioEmitter>;
-			audioequalizer: React.InstanceProps<AudioEqualizer>;
-			audiofader: React.InstanceProps<AudioFader>;
-			audioflanger: React.InstanceProps<AudioFlanger>;
-			audiolistener: React.InstanceProps<AudioListener>;
-			audiopitchshifter: React.InstanceProps<AudioPitchShifter>;
-			audioplayer: React.InstanceProps<AudioPlayer>;
-			audioreverb: React.InstanceProps<AudioReverb>;
-			ballsocketconstraint: React.InstanceProps<BallSocketConstraint>;
-			beam: React.InstanceProps<Beam>;
-			billboardgui: React.InstanceProps<BillboardGui>;
-			blockmesh: React.InstanceProps<BlockMesh>;
-			bloomeffect: React.InstanceProps<BloomEffect>;
-			blureffect: React.InstanceProps<BlurEffect>;
-			bodyangularvelocity: React.InstanceProps<BodyAngularVelocity>;
-			bodycolors: React.InstanceProps<BodyColors>;
-			bodyforce: React.InstanceProps<BodyForce>;
-			bodygyro: React.InstanceProps<BodyGyro>;
-			bodyposition: React.InstanceProps<BodyPosition>;
-			bodythrust: React.InstanceProps<BodyThrust>;
-			bodyvelocity: React.InstanceProps<BodyVelocity>;
-			bone: React.InstanceProps<Bone>;
-			boolvalue: React.InstanceProps<BoolValue>;
-			boxhandleadornment: React.InstanceProps<BoxHandleAdornment>;
-			brickcolorvalue: React.InstanceProps<BrickColorValue>;
-			buoyancysensor: React.InstanceProps<BuoyancySensor>;
-			camera: React.InstanceProps<Camera>;
-			canvasgroup: React.InstanceProps<CanvasGroup>;
-			cframevalue: React.InstanceProps<CFrameValue>;
-			charactermesh: React.InstanceProps<CharacterMesh>;
-			chorussoundeffect: React.InstanceProps<ChorusSoundEffect>;
-			clickdetector: React.InstanceProps<ClickDetector>;
-			climbcontroller: React.InstanceProps<ClimbController>;
-			clouds: React.InstanceProps<Clouds>;
-			color3value: React.InstanceProps<Color3Value>;
-			colorcorrectioneffect: React.InstanceProps<ColorCorrectionEffect>;
-			compressorsoundeffect: React.InstanceProps<CompressorSoundEffect>;
-			conehandleadornment: React.InstanceProps<ConeHandleAdornment>;
-			configuration: React.InstanceProps<Configuration>;
-			controllermanager: React.InstanceProps<ControllerManager>;
-			controllerpartsensor: React.InstanceProps<ControllerPartSensor>;
-			cornerwedgepart: React.InstanceProps<CornerWedgePart>;
-			curveanimation: React.InstanceProps<CurveAnimation>;
-			cylinderhandleadornment: React.InstanceProps<CylinderHandleAdornment>;
-			cylindermesh: React.InstanceProps<CylinderMesh>;
-			cylindricalconstraint: React.InstanceProps<CylindricalConstraint>;
-			decal: React.InstanceProps<Decal>;
-			depthoffieldeffect: React.InstanceProps<DepthOfFieldEffect>;
-			distortionsoundeffect: React.InstanceProps<DistortionSoundEffect>;
-			doubleconstrainedvalue: React.InstanceProps<DoubleConstrainedValue>;
-			dragdetector: React.InstanceProps<DragDetector>;
-			dragger: React.InstanceProps<Dragger>;
-			echosoundeffect: React.InstanceProps<EchoSoundEffect>;
-			editableimage: React.InstanceProps<EditableImage>;
-			editablemesh: React.InstanceProps<EditableMesh>;
-			equalizersoundeffect: React.InstanceProps<EqualizerSoundEffect>;
-			eulerrotationcurve: React.InstanceProps<EulerRotationCurve>;
-			facecontrols: React.InstanceProps<FaceControls>;
-			fire: React.InstanceProps<Fire>;
-			flangesoundeffect: React.InstanceProps<FlangeSoundEffect>;
-			floatcurve: React.InstanceProps<FloatCurve>;
-			floorwire: React.InstanceProps<FloorWire>;
-			folder: React.InstanceProps<Folder>;
-			forcefield: React.InstanceProps<ForceField>;
-			frame: React.InstanceProps<Frame>;
-			groundcontroller: React.InstanceProps<GroundController>;
-			handles: React.InstanceProps<Handles>;
-			highlight: React.InstanceProps<Highlight>;
-			hingeconstraint: React.InstanceProps<HingeConstraint>;
-			hole: React.InstanceProps<Hole>;
-			humanoid: React.InstanceProps<Humanoid>;
-			humanoidcontroller: React.InstanceProps<HumanoidController>;
-			humanoiddescription: React.InstanceProps<HumanoidDescription>;
-			ikcontrol: React.InstanceProps<IKControl>;
-			imagebutton: React.InstanceProps<ImageButton>;
-			imagehandleadornment: React.InstanceProps<ImageHandleAdornment>;
-			imagelabel: React.InstanceProps<ImageLabel>;
-			intconstrainedvalue: React.InstanceProps<IntConstrainedValue>;
-			intvalue: React.InstanceProps<IntValue>;
-			keyframe: React.InstanceProps<Keyframe>;
-			keyframemarker: React.InstanceProps<KeyframeMarker>;
-			keyframesequence: React.InstanceProps<KeyframeSequence>;
-			linearvelocity: React.InstanceProps<LinearVelocity>;
-			lineforce: React.InstanceProps<LineForce>;
-			linehandleadornment: React.InstanceProps<LineHandleAdornment>;
-			localizationtable: React.InstanceProps<LocalizationTable>;
-			localscript: React.InstanceProps<LocalScript>;
-			markercurve: React.InstanceProps<MarkerCurve>;
-			materialvariant: React.InstanceProps<MaterialVariant>;
-			model: React.InstanceProps<Model>;
-			motor: React.InstanceProps<Motor>;
-			motor6d: React.InstanceProps<Motor6D>;
-			nocollisionconstraint: React.InstanceProps<NoCollisionConstraint>;
-			numberpose: React.InstanceProps<NumberPose>;
-			numbervalue: React.InstanceProps<NumberValue>;
-			objectvalue: React.InstanceProps<ObjectValue>;
-			pants: React.InstanceProps<Pants>;
-			part: React.InstanceProps<Part>;
-			particleemitter: React.InstanceProps<ParticleEmitter>;
-			pitchshiftsoundeffect: React.InstanceProps<PitchShiftSoundEffect>;
-			planeconstraint: React.InstanceProps<PlaneConstraint>;
-			pointlight: React.InstanceProps<PointLight>;
-			pose: React.InstanceProps<Pose>;
-			prismaticconstraint: React.InstanceProps<PrismaticConstraint>;
-			proximityprompt: React.InstanceProps<ProximityPrompt>;
-			rayvalue: React.InstanceProps<RayValue>;
-			reverbsoundeffect: React.InstanceProps<ReverbSoundEffect>;
-			rigidconstraint: React.InstanceProps<RigidConstraint>;
-			rocketpropulsion: React.InstanceProps<RocketPropulsion>;
-			rodconstraint: React.InstanceProps<RodConstraint>;
-			ropeconstraint: React.InstanceProps<RopeConstraint>;
-			rotationcurve: React.InstanceProps<RotationCurve>;
-			screengui: React.InstanceProps<ScreenGui>;
-			script: React.InstanceProps<Script>;
-			scrollingframe: React.InstanceProps<ScrollingFrame>;
-			seat: React.InstanceProps<Seat>;
-			selectionbox: React.InstanceProps<SelectionBox>;
-			selectionsphere: React.InstanceProps<SelectionSphere>;
-			shirt: React.InstanceProps<Shirt>;
-			shirtgraphic: React.InstanceProps<ShirtGraphic>;
-			sky: React.InstanceProps<Sky>;
-			smoke: React.InstanceProps<Smoke>;
-			sound: React.InstanceProps<Sound>;
-			soundgroup: React.InstanceProps<SoundGroup>;
-			sparkles: React.InstanceProps<Sparkles>;
-			spawnlocation: React.InstanceProps<SpawnLocation>;
-			specialmesh: React.InstanceProps<SpecialMesh>;
-			spherehandleadornment: React.InstanceProps<SphereHandleAdornment>;
-			spotlight: React.InstanceProps<SpotLight>;
-			springconstraint: React.InstanceProps<SpringConstraint>;
-			stringvalue: React.InstanceProps<StringValue>;
-			stylederive: React.InstanceProps<StyleDerive>;
-			stylelink: React.InstanceProps<StyleLink>;
-			stylerule: React.InstanceProps<StyleRule>;
-			stylesheet: React.InstanceProps<StyleSheet>;
-			sunrayseffect: React.InstanceProps<SunRaysEffect>;
-			surfaceappearance: React.InstanceProps<SurfaceAppearance>;
-			surfacegui: React.InstanceProps<SurfaceGui>;
-			surfacelight: React.InstanceProps<SurfaceLight>;
-			surfaceselection: React.InstanceProps<SurfaceSelection>;
-			swimcontroller: React.InstanceProps<SwimController>;
-			textbox: React.InstanceProps<TextBox>;
-			textbutton: React.InstanceProps<TextButton>;
-			textlabel: React.InstanceProps<TextLabel>;
-			texture: React.InstanceProps<Texture>;
-			tool: React.InstanceProps<Tool>;
-			torque: React.InstanceProps<Torque>;
-			torsionspringconstraint: React.InstanceProps<TorsionSpringConstraint>;
-			trail: React.InstanceProps<Trail>;
-			tremolosoundeffect: React.InstanceProps<TremoloSoundEffect>;
-			trusspart: React.InstanceProps<TrussPart>;
-			uiaspectratioconstraint: React.InstanceProps<UIAspectRatioConstraint>;
-			uicorner: React.InstanceProps<UICorner>;
-			uidragdetector: React.InstanceProps<UIDragDetector>;
-			uiflexitem: React.InstanceProps<UIFlexItem>;
-			uigradient: React.InstanceProps<UIGradient>;
-			uigridlayout: React.InstanceProps<UIGridLayout>;
-			uilistlayout: React.InstanceProps<UIListLayout>;
-			uipadding: React.InstanceProps<UIPadding>;
-			uipagelayout: React.InstanceProps<UIPageLayout>;
-			uiscale: React.InstanceProps<UIScale>;
-			uisizeconstraint: React.InstanceProps<UISizeConstraint>;
-			uistroke: React.InstanceProps<UIStroke>;
-			uitablelayout: React.InstanceProps<UITableLayout>;
-			uitextsizeconstraint: React.InstanceProps<UITextSizeConstraint>;
-			universalconstraint: React.InstanceProps<UniversalConstraint>;
-			vector3curve: React.InstanceProps<Vector3Curve>;
-			vector3value: React.InstanceProps<Vector3Value>;
-			vectorforce: React.InstanceProps<VectorForce>;
-			vehiclecontroller: React.InstanceProps<VehicleController>;
-			vehicleseat: React.InstanceProps<VehicleSeat>;
-			velocitymotor: React.InstanceProps<VelocityMotor>;
-			videoframe: React.InstanceProps<VideoFrame>;
-			viewportframe: React.InstanceProps<ViewportFrame>;
-			wedgepart: React.InstanceProps<WedgePart>;
-			weld: React.InstanceProps<Weld>;
-			weldconstraint: React.InstanceProps<WeldConstraint>;
-			wire: React.InstanceProps<Wire>;
-			wireframehandleadornment: React.InstanceProps<WireframeHandleAdornment>;
-			worldmodel: React.InstanceProps<WorldModel>;
-			wraplayer: React.InstanceProps<WrapLayer>;
-			wraptarget: React.InstanceProps<WrapTarget>;
-		}
-	}
-}
